cuda/matrix/csr_kernels.cu

```diff
@@ -86,6 +86,9 @@ constexpr int wsize = config::warp_size;
  */
 using compiled_kernels = syn::value_list<int, 3, 4, 6, 7, 8, 12, 14>;
 
+using classical_kernels =
+    syn::value_list<int, config::warp_size, 32, 16, 8, 4, 2, 1>;
```

Comment:
Just for completeness, did you try on one machine/case to sort these in reverse order and confirm that performance was worst?

Reply:
No.
I would like to find the warp_size is less or equal to the number of max nnz per row.
If reverse it under this condition, it always takes 1 as the warp_size.

If we reverse it and find the warp_size is larger or equal to the number of max nnz per row, I guess the performance is similar because use one more warp not one more addition.